lib/src/base_request_builder.dart

```diff
@@ -16,4 +16,31 @@ abstract class BaseRequestBuilder {
 
   /// Url template to use to build the URL for the current request builder.
   String urlTemplate;
+
+  /// Clones the current request builder.
+  T clone();
+
+  /// Clones the current request builder using [clone] and applies the provided
+  /// parameters.
+  T copyWith({
```

Comment:
The value of the WithUrl method is to provide the ability to use the fluent API with arbitrary URLs returned by the service.
While this implementation is more flexible, it diverges from the other languages, and strays a bit in the general intent.
I'd encourage to align with the other languages for now and if we want to make a general improvement like this one, we'll coordinate it across languages

Reply:
Okay, so what's the best course of action now? Close this PR and maybe add something like `withUrl` later?

Reply:
just renaming the method here, changing the parameters would be perfect I think.

Reply:
Ok, done